src/hooks/screen/useAnonymousChatScreenHook.ts

```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line import/no-unresolved
+import 'react-native-get-random-values';
+
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import UseAnonymousChatScreenHook from '../../../types/hooks/screens/useAnonymousChatScreenHook.types';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {getAnonymousUserId} from '../../utils/users';
+import {getUserId} from '../../utils/token';
+import {randomString} from '../../utils/string/StringUtils';
+
+function useAnonymousChatScreenHook(): UseAnonymousChatScreenHook {
+  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
+
+  const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
+  const initChatData = async () => {
+    if (!localDb && !selectedChannel) return;
+    const myUserId = await getUserId();
+    const myAnonymousId = await getAnonymousUserId();
+    const data = (await ChatSchema.getAll(
+      localDb,
+      selectedChannel?.id,
+      myUserId,
+      myAnonymousId
+    )) as ChatSchema[];
+    setChats(data);
+  };
+
+  const sendChat = async (
+    message: string = randomString(20),
+    iteration = 0,
+    sendingChatSchema: ChatSchema = null
+  ) => {
+    if (iteration > 5) {
+      SimpleToast.show("Can't send message, please check your connection");
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error message "Can't send message, please check your connection" is hardcoded. It would be better to move this to a constants file or use a localization library if you are planning to support multiple languages in the future.
```diff
-      SimpleToast.show("Can't send message, please check your connection");
+      SimpleToast.show(ERROR_MESSAGES.CANT_SEND_MESSAGE);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->